### PR TITLE
Changing the type of the asset description from String to Vec<u8>

### PR DESCRIPTION
**Pull request by vivek-arte**
This PR makes the changes that are required after the merging of https://github.com/QED-it/orchard/pull/113

### CODE REVIEW

File: zcash_primitives/src/transaction/builder.rs [hunk 1]
```diff
@@ -1455,17 +1455,23 @@ mod tests {
     fn init_issuance_bundle_with_finalization() {
         let (mut builder, iak, _) = prepare_zsa_test();
 
-        let asset = "asset_desc".to_string();
+        let asset_desc: &[u8] = b"asset_desc";
```
[the remark this answers is not in the document]
Comment (PaulLaux) [reply zcash_primitives/src/transaction/builder.rs:1458]: Later, it makes more scene to use `Vec` as a slice then the opposite.

**Review by PaulLaux — COMMENTED** [review]
Added some minor changes

File: Cargo.toml [hunk 2]
```diff
@@ -58,7 +58,7 @@ sapling = { package = "sapling-crypto", version = "0.1.3" }
 
 # - Orchard
 nonempty = "0.7"
-orchard = { version = "0.8.0", default-features = false, git = "https://github.com/QED-it/orchard", branch = "zsa1" }
+orchard = { version = "0.8.0", default-features = false, git = "https://github.com/QED-it/orchard", branch = "asset_desc_as_bytes" }
```
[the remark this answers is not in the document]
Comment (vivek-arte) [reply Cargo.toml:61]: To do: Waiting for Orchard PR to merge

File: zcash_primitives/src/transaction/builder.rs [hunk 3]
```diff
@@ -1455,17 +1455,23 @@ mod tests {
...
-            .init_issuance_bundle::<FeeError>(iak, asset.clone(), None)
+            .init_issuance_bundle::<FeeError>(iak, Vec::from(asset_desc), None)
             .unwrap();
 
         let tx = builder.mock_build_no_fee(OsRng).unwrap().into_transaction();
         let bundle = tx.issue_bundle().unwrap();
 
         assert_eq!(bundle.actions().len(), 1, "There should be only one action");
-        let action = bundle.get_action(asset).unwrap();
+        let actions_vec = bundle.get_actions_by_desc(asset_desc);
```
[the remark this answers is not in the document]
Comment (vivek-arte) [reply zcash_primitives/src/transaction/builder.rs:1468]: Makes sense, refactored

File: zcash_primitives/src/transaction/builder.rs [hunk 4]
```diff
@@ -1511,32 +1519,34 @@ mod tests {
...
-            .add_recipient::<FeeError>(asset.clone(), address, NoteValue::from_raw(21))
+            .add_recipient::<FeeError>(asset_desc, address, NoteValue::from_raw(21))
             .unwrap();
 
         let binding = builder.mock_build_no_fee(OsRng).unwrap().into_transaction();
         let bundle = binding.issue_bundle().unwrap();
 
         assert_eq!(bundle.actions().len(), 1, "There should be only one action");
-        let action = bundle.get_action(asset).unwrap();
+        let actions_vec = bundle.get_actions_by_desc(asset_desc);
```
[the remark this answers is not in the document]
Comment (vivek-arte) [reply zcash_primitives/src/transaction/builder.rs:1542]: Refactored

File: zcash_primitives/src/transaction/builder.rs [hunk 5]
```diff
@@ -1554,34 +1564,36 @@ mod tests {
...
+            .add_recipient::<FeeError>(asset_desc_2, address, NoteValue::from_raw(21))
             .unwrap();
 
         let binding = builder.mock_build_no_fee(OsRng).unwrap().into_transaction();
         let bundle = binding.issue_bundle().unwrap();
 
         assert_eq!(bundle.actions().len(), 2, "There should be 2 actions");
 
-        let action = bundle.get_action(asset1).unwrap();
+        let actions_vec = bundle.get_actions_by_desc(asset_desc_1);
```
[the remark this answers is not in the document]
Comment (vivek-arte) [reply zcash_primitives/src/transaction/builder.rs:1589]: Refactored

File: zcash_primitives/src/transaction/builder.rs [hunk 6]
```diff
@@ -1593,12 +1605,14 @@ mod tests {
             "Incorrect notes sum"
         );
 
-        let action2 = bundle.get_action(asset2).unwrap();
+        let actions_vec_2 = bundle.get_actions_by_desc(asset_desc_2);
```
[the remark this answers is not in the document]
Comment (vivek-arte) [reply zcash_primitives/src/transaction/builder.rs:1608]: Refactored

File: zcash_primitives/src/transaction/builder.rs [hunk 7]
```diff
@@ -1554,34 +1564,36 @@ mod tests {
     fn add_issuance_different_asset() {
         let (mut builder, iak, address) = prepare_zsa_test();
 
-        let asset1 = "asset_desc".to_string();
-        let asset2 = "asset_desc_2".to_string();
+        let asset_desc_1: &[u8] = b"asset_desc";
+        let asset_desc_2: &[u8] = b"asset_desc_2";
```
[the remark this answers is not in the document]
Comment (vivek-arte) [reply zcash_primitives/src/transaction/builder.rs:1568]: Done

File: zcash_primitives/src/transaction/builder.rs [hunk 8]
```diff
@@ -1511,32 +1519,34 @@ mod tests {
     fn add_issuance_same_asset() {
         let (mut builder, iak, address) = prepare_zsa_test();
 
-        let asset = "asset_desc".to_string();
+        let asset_desc: &[u8] = b"asset_desc";
```
[the remark this answers is not in the document]
Comment (vivek-arte) [reply zcash_primitives/src/transaction/builder.rs:1522]: Done

File: zcash_primitives/src/transaction/builder.rs [hunk 9]
```diff
@@ -1475,12 +1481,12 @@ mod tests {
     fn init_issuance_bundle_without_finalization() {
         let (mut builder, iak, address) = prepare_zsa_test();
 
-        let asset = "asset_desc".to_string();
+        let asset_desc: &[u8] = b"asset_desc";
```
[the remark this answers is not in the document]
Comment (vivek-arte) [reply zcash_primitives/src/transaction/builder.rs:1484]: Done

File: zcash_primitives/src/transaction/builder.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (vivek-arte) [reply zcash_primitives/src/transaction/builder.rs:1458]: Done

**Review by PaulLaux — APPROVED** [review]

**Pull request merged**